modules/travel_pay/app/services/travel_pay/appointments_client.rb

```diff
@@ -23,7 +23,11 @@ def get_all_appointments(veis_token, btsss_token, params = {})
       correlation_id = SecureRandom.uuid
       Rails.logger.debug(message: 'Correlation ID', correlation_id:)
 
-      query_path = URI::HTTPS.build(path: '/api/v1.1/appointments', query: params.to_query)
+      query_path = if params.empty?
+                     '/api/v1.1/appointments'
+                   else
+                     "/api/v1.1/appointments?#{params.to_query}"
+                   end
```

Comment:
After playing around a little bit, I think you were close... does this work (note the `request_uri`)?

```ruby
query_path = URI::HTTPS.build(path: '/api/v1.1/appointments', query: params.to_query)

connection(server_url: btsss_url).get(query_path.request_uri) do |req|
  # ...
end
```

[source](https://ruby-doc.org/stdlib-2.5.1/libdoc/uri/rdoc/URI/HTTP.html#method-i-request_uri-label-Description)

Reply:
After testing various versions of using `URI::HTTPS` it seems like this is not going to provide what we need it to.